nixops/backends/__init__.py

```diff
@@ -273,12 +275,18 @@ def send_keys(self) -> None:
             return
         if self.store_keys_on_machine:
             return
-        for k, opts in self.get_keys().items():
+
```

Comment:
When I tried this, all the threads tried to open the control connection at once:

```
flexo.......> uploading key ‘wireguard.nix’...
flexo.......> uploading key ‘wireguard1’...
flexo.......> uploading key ‘wireguard2’...
flexo.......> uploading key ‘wireguard3’...
flexo.......> uploading key ‘wireguard4’...
flexo.......> uploading key ‘wireguard5’...
flexo.......> uploading key ‘wireguard6’...
flexo.......> sign_and_send_pubkey: signing failed for RSA "/home/grahamc/.ssh/id_rsa" from agent: agent refused operation
flexo.......> mux_client_request_session: read from master failed: Broken pipe
flexo.......> mux_client_request_session: read from master failed: Broken pipe
flexo.......> mux_client_request_session: read from master failed: Broken pipe
flexo.......> mux_client_request_session: read from master failed: Broken pipe
flexo.......> uploading key ‘wireguard7’...
flexo.......> uploading key ‘wireguard8’...
flexo.......> uploading key ‘wireguard9’...
```

Preflighting the connection lets the control socket open, and then proceed:

```suggestion
        # preflight the connection
        self.run_command("true")
```